### ISSUE TITLE
After server restart, SwellRT returns 403 Forbidden to sign-in

### ISSUE THREAD

**Issue by atd** [P2Pvalue/swellrt#96]
After a SwellRT restart, sign-in calls return 403 Forbidden, despite credentials are sent correctly

```
POST /auth/signin?r=none HTTP/1.1
Host: swellrt:9898
Connection: keep-alive
Content-Length: 46
Origin: http://swellrt:8000
User-Agent: Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/46.0.2490.71 Safari/537.36
Content-Type: application/x-www-form-urlencoded
Accept: */*
Referer: http://swellrt:8000/
Accept-Encoding: gzip, deflate
Accept-Language: en-US,en;q=0.8,es;q=0.6
Cookie: WSESSIONID=nji1q3jrszrzkr7d8xvifd5f

HTTP/1.1 403 Forbidden
Access-Control-Allow-Origin: http://swellrt:8000
Access-Control-Allow-Credentials: true
Access-Control-Expose-Headers: 
Content-Type: text/plain; charset=UTF-8
Content-Length: 15
Server: Jetty(9.1.z-SNAPSHOT)

login forbidden
```

**Comment by pablojan** [comment]
I can't reproduce. Plz provide more context info... e.g.

What is the current state of the client's session when the server goes down?
Does the sign-in call use the same credentials?

**Comment by atd** [comment]
Sure:

> What is the current state of the client's session when the server goes down?

Logged in

> Does the sign-in call use the same credentials?

Yes same credentials.

The key point is removing session data from mongodb, so session does not exist anymore. Sorry for not giving enough details.

**Comment by pablojan** [comment]
This issue happens on docker container restart (no just on jetty restart). 

During docker restart, filesystem folder containing persisted jetty sessions (see swellrt's config property `sessions_store_directory = _sessions`) is deleted, hence persisted sessions are not restored.

Suggested solution is to configure the session store directory as a external folder to the container.

**Comment by atd** [comment]
I am implementing this, but I think I would be better if they were stored in the mongodb with the rest of the data.

Maybe this is asking too much? :grimacing:

**Comment by pablojan** [comment]
Wow, session manager is part of  the Jetty server.., it sounds horrible!

2016-02-09 17:36 GMT+01:00 Antonio Tapiador <notifications@github.com>:

**Comment by atd** [comment]
I think that current solution has problems from the sysadmin point of view, but... I am not going to push forward :smile:

**Comment by atd** [comment]
Closed via 82a128c0ce4a2c0d73e48498f054741bf417bdbe
